src/borg/archive.py

```diff
@@ -468,6 +470,8 @@ def save(self, name=None, comment=None, timestamp=None, additional_metadata=None
         data = self.key.pack_and_authenticate_metadata(metadata.as_dict(), context=b'archive')
         self.id = self.key.id_hash(data)
         self.cache.add_chunk(self.id, Chunk(data), self.stats)
+        self.repository.async_wait()
```

Comment:
I don't think that's a good idea, since it has the same problem the fixed preload window has, i.e. it leads to poor utilization of bandwidth if latency is high.